### PR TITLE
Feature/orca 678 db_deploy: Add collection_id for recovery jobs

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Added collection_id to recovery status tables.

Addresses [ORCA-678: db_deploy: Add collection_id for recovery jobs](https://bugs.earthdata.nasa.gov/browse/ORCA-678)

## Changes

* Added collection_id column to recovery_files and recovery_jobs
* Added migration to add column and populate with data from `granules` table
* Added tests

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Code run against AWS deployment.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ and includes an additional section for migration notes.
 - *ORCA-597*
   - Server access logging is now enabled for graphql application load balancer.
 - *ORCA-614*, *ORCA-428* Moved some Internal Reconciliation functionality to GraphQL
+- *ORCA-679* Updated area in recovery where granule ID was treated as a globally unique key. Per Cumulus updates, uniqueness is no granule ID plus collection ID.
```
Comment (AndrewDorn) [inline CHANGELOG.md:28]: This is the Epic number. Opted to create a side-branch and merge into that to keep develop functional.

**Review by rizbihassan — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply CHANGELOG.md:28]: Fixed

File: tasks/db_deploy/migrations/migrate_versions_6_to_7/migrate.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"""
+Name: migrate.py
+
+Description: Migrates the ORCA schema from version 5 to version 6.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/db_deploy/migrations/migrate_versions_6_to_7/migrate.py:4]: Fixed

File: tasks/db_deploy/migrations/migrate_versions_6_to_7/migrate_sql.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+"""
+Name: migrate_sql.py
+
+Description: All the SQL used for creating and migrating the ORCA schema to version 6.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/db_deploy/migrations/migrate_versions_6_to_7/migrate_sql.py:4]: Fixed

File: tasks/db_deploy/test/unit_tests/migrations/migrate_versions_6_to_7/test_migrate_sql_v7.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+"""
+Name: test_migrate_sql_v6.py
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/db_deploy/test/unit_tests/migrations/migrate_versions_6_to_7/test_migrate_sql_v7.py:2]: Fixed

**Review by rizbihassan — APPROVED** [review]

**Pull request merged**